tests/testthat/test_smaller.R

```diff
@@ -347,4 +409,71 @@ test_that("Returns correct subsetted object", {
 
 # TODO: Add tests for ligerToSeurat and seuratToLiger functions 
 # after including functionality related to new cell.data slot 
+seurat_from_liger <- ligerToSeurat(ligex)
+test_that("Returns correct Seurat object", {
+  expect_is(seurat_from_liger, c('Seurat'))
+  expect_equal(dim(seurat_from_liger@assays$RNA), c(3294, 494))
+  expect_equal(as.numeric(unname(seurat_from_liger@assays$RNA[11,11])), 5.172973, tolerance = 1e-6)
+  expect_equal(as.numeric(unname(seurat_from_liger@assays$RNA[505,101])), 3.423402, tolerance = 1e-6)
+  expect_equal(dim(seurat_from_liger@reductions$tsne), c(494,2))
+  expect_equal(dim(seurat_from_liger@reductions$inmf), c(494,15))
+  expect_equal(unname(seurat_from_liger@reductions$inmf@feature.loadings["RPS18",15]), 2.363831
+               , tolerance = 1e-6)
+})
+
+ligex_from_seurat <- seuratToLiger(pbmc.seurat,combined.seurat = T, assays.use = c("RNA"))
+
+
+rm(seurat_from_liger, liger_from_seurat)
+####################################################################################
+context("Metrics")
+
+#calcDatasetSpecificity
+dataset_spec <- calcDatasetSpecificity(ligex, do.plot = F)
+test_that("Returns correct dataset specificity", {
+  expect_equal(dataset_spec[[1]][1:5], c(96.31750, 135.68717, 108.94444, 89.92569, 84.49224),
+               tolerance = 1e-6)
+  expect_equal(dataset_spec[[2]][6:10], c(100.91134, 112.07151, 123.82733, 77.75074, 88.71075),
+               tolerance = 1e-6)
+  expect_equal(dataset_spec[[3]][11:15], c(0.03951941, -0.51374525, -0.70291549, -0.06761655,
+               -0.79589711),tolerance = 1e-6)
+})
+
+#calcAgreement
+#calcAlignment
+agreement <- calcAgreement(ligex)
+alignment <- calcAlignment(ligex, clusters.use = c(1,2))
+test_that("Returns correct agreement and alignment", {
+  expect_equal(agreement, 0.1974746, tolerance = 1e-6)
+  expect_equal(alignment, 0.9990196, tolerance = 1e-6)
+})
+
+#calcAlignmentPerCluster
+#not working for some reason, despite alignment itself working
+
+cluster_levels <- as.vector(ligex@clusters)
```

Comment:
This block of code seems to be duplicated directly below -- think it only needs to be there once!